### ISSUE TITLE
On Accurate Evaluation of GANs for Language Generation

### ISSUE THREAD

**Issue by howardyclo** [howardyclo/papernotes#21]
### Metadata
Authors: Stanislau Semeniuta, Aliaksei Severyn, Sylvain Gelly
Organization: Google AI
Conference: NIPS 2018
Paper: https://arxiv.org/pdf/1806.04936.pdf

**Comment by howardyclo** [comment]
### Summary
This paper presents a systematic re-evaluation for a number of recently purposed GAN-based text generation models. They argued that N-gram based metrics (e.g., BLEU, Self-BLEU) is not sensitive to semantic deterioration of generated texts and propose alternative metrics that better capture the quality and diversity of the generated samples. Their main finding is that, neither of the considered models performs convincingly better than the conventional Language Model (LM). Furthermore, when performing a hyperparameter search, they consistently find that adversarial learning hurts performance, further indicating that the Language Model is still a hard-to-beat model.

### Motivations for GAN-based Text Generation Models
- Exposure bias problem (conditioning on ground truth while training v.s. on its own imperfect predictions while inference) and objective mismatch (negative log-likehood training vs task-specific metric evaluation) exist in conventional neural language model.
- GAN-based text generation mitigates both issues described above.

### Motivations for This Paper
- GAN-based text generation is still challenging.
- One of the many challenges that slows down the progress is the lack of proper evaluation, which is a largely unsolved problem and is an active area of research. 

### Types fo Benchmarked Models
- Continuous WGAN-GP: Treat a sequence of tokens as a one-dimensional signal and directly backpropagate from the discriminator. For RNN-based generator G, yhey feed the entire softmax output as the input for the next step of G.
- **To be continued**

### Related Work
- [Neural Text Generation: Past, Present and Beyond](https://arxiv.org/pdf/1803.07133v1.pdf) by Lu et al., 2018
- [Eval all, trust a few, do wrong to none: Comparing sentence generation models](https://arxiv.org/pdf/1804.07972.pdf) by Cífka et al., 2018.

**Comment by xpqiu** [comment]
Dear howardyclo, 
Thanks for your great comments. 
I just suggest another BLEU based metric to better capture the quality and diversity of the generated samples.  The overall BLEU score consists of two parts: a forward BLEU aims to measure the precision (quality) of the generator, while backward BLEU aims to measure the recall (diversity) of the generator. 
The details is shown in the bellowing paper.
“Toward Diverse Text Generation with Inverse Reinforcement Learning”
 https://arxiv.org/pdf/1804.11258.pdf

**Comment by howardyclo** [comment]
@xpqiu Hi Xipeng
Actually, your paper is currently in my paper reading pending list! Ha!
I will soon get to your paper and write a summary about your it.
It would be great if you can help me to review the summary, thanks!
